python_modules/dagster/dagster/core/events/transform.py

```diff
@@ -0,0 +1,49 @@
+from enum import Enum
+from collections import namedtuple
+
+from dagster import check
+from dagster.core.execution_context import SystemTransformExecutionContext
+from dagster.core.events.logging import EventType
+from dagster.core.events import DagsterEvent
+
+
+class TransformEventType(Enum):
+    STEP_MATERIALIZATION = 'STEP_MATERIALIZATION'
+
+
+class TransformEvent(namedtuple('_TransformEvent', 'event_type_value name path'), DagsterEvent):
```

Comment:
I actually don't see why this shouldn't be a general step event. Any type of event can produce a materialization that will outlive the computation.